src/engine/controls/loopingcontrol.cpp

```diff
@@ -1228,15 +1231,20 @@ void LoopingControl::trackLoaded(TrackPointer pNewTrack) {
 
 void LoopingControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
     clearActiveBeatLoop();
-    m_pBeats = pBeats;
-    if (m_pBeats) {
-        LoopInfo loopInfo = m_loopInfo.getValue();
-        if (loopInfo.startPosition.isValid() && loopInfo.endPosition.isValid()) {
-            double loaded_loop_size = findBeatloopSizeForLoop(
-                    loopInfo.startPosition, loopInfo.endPosition);
-            if (loaded_loop_size != -1) {
-                m_pCOBeatLoopSize->setAndConfirm(loaded_loop_size);
-            }
+    if (pBeats) {
+        m_pBeats = pBeats;
+        m_trueTrackBeats = true;
+    } else {
+        m_pBeats = getFake60BpmBeats();
+        m_trueTrackBeats = false;
+    }
+    // TODO All "if (m_pBeats)" checks are now obsolete actually...
```

Comment:
Than these `if (m_pBeats)` need to be removed from the code

Reply:
I have the impression that it is used as `if (m_pTrack)`.
I'll take a look.

Reply:
If refined updateBeats() so we now have a Beats nullptr again if no track is loaded, and all `if (m_pBeats)` checks can stay.